src/projections.jl

```diff
@@ -446,9 +522,97 @@ References:
 (https://stanford.edu/~boyd/papers/cone_prog_refine.html)
 by Enzo Busseti, Walaa M. Moursi, and Stephen Boyd
 """
-function projection_gradient_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.DualExponentialCone) where {T}
+function projection_gradient_on_set(d::DefaultDistance, v::AbstractVector{T}, ::MOI.DualExponentialCone) where {T}
+    # from Moreau decomposition: x = P_K(x) + P_-K*(x)
+    return I - projection_gradient_on_set(d, -v, MOI.ExponentialCone())
+end
+
+"""
+    projection_gradient_on_set(d::DefaultDistance, v::AbstractVector{T}, ::MOI.PowerCone) where {T}
+
+derivative of projection of vector `v` on the power cone
+i.e. `K = {(x,y,z) | x^a * y^(1-a) >= |z|, x>=0, y>=0}`.
+
+References:
+* [Differential properties of Euclidean projection onto power cone]
+(https://link.springer.com/article/10.1007/s00186-015-0514-0), Theorem 3.1
+"""
+function projection_gradient_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.PowerCone) where {T}
+    _check_dimension(v, s)
+
+    if _in_pow_cone(v, s)
+        return Matrix{T}(I, 3, 3)
+    end
+    if _in_pow_cone(-v, MOI.dual_set(s))
+        # if in polar cone Ko = -K*
+        return zeros(T, 3, 3)
+    end
+    if abs(v[3]) <= 1e-8
+        return _pow_cone_∇proj_case_3(v, s)
+    end
+
+    x, y, z = v
+    α = s.exponent
+    # # This handles the case where v[3]/norm(v) is small.
+    #   Phi(eps()) ≈ -r
+    #   Phi_prod(x, α, z, eps()) ≈ x + |x| = 2*max(x, 0)
+    #   => this gets to case 3 in the limit
+    # if (x < 0 || y < 0) && abs(z) <= 1e-2*norm(v)*(0.5 - abs(0.5 - α))
+    #     r = eps()
+    # else
+    #     r = _solve_system_pow_cone(v, s; arg=true)
+    # end
+
+    r = _solve_system_pow_cone(v, s; arg=true)
+    za = abs(z)
+    gx = sqrt(x^2 + 4*α*r*(za - r))
+    gy = sqrt(y^2 + 4*(1-α)*r*(za - r))
+    fx = 0.5*(x + gx)
+    fy = 0.5*(y + gy)
+
+    β = 1-α
+    K = -(α*x/gx + β*y/gy)
+    L = 2*(za - r) / (za + (za - 2r) * -K)
+    J_ii(w, γ, g) = 0.5 + w/2g + γ^2*(za - 2r)*r*L/g^2
+    J_ij = α*β*(za - 2r)*r*L/(gx*gy)
+    J = [
+        J_ii(x, α, gx)      J_ij                sign(z)*α*r*L/gx;
+        J_ij                J_ii(y, β, gy)      sign(z)*β*r*L/gy;
+        sign(z)*α*r*L/gx    sign(z)*β*r*L/gy    r/za*(1+K*L)
+    ]
+    return J
+end
+
+function _pow_cone_∇proj_case_3(v::AbstractVector{T}, s::MOI.PowerCone) where {T}
+    x = [v[1]; v[2]]
+    I(t) = t > 0 ? 1 : 0
```

Comment:
You can replace I(t) by `t <= 0` directly in the whole function body, since a Bool is a subtype of integer